### DISCUSSION TITLE
Box as input

### DISCUSSION THREAD

**Discussion by zkuzmic** [twilio-labs/paste#2131] [Ideas] [answered]
Hello!

I'm trying to build a styled input using `Box` and am running into a type error that the `value` prop doesn't exist:

<img width="341" alt="Screen Shot 2022-01-12 at 2 12 38 PM" src="https://user-images.githubusercontent.com/1144042/149214527-2ccc1620-794c-4e73-a896-c9515ae6fdaf.png">

Is this expected/desired? If not, I'm wondering if it makes sense to make the `BoxElementProps` a discriminated union that allows `value` when the `as` prop is "input".

I'd be happy to open a PR, just wanted to check in here first.

**Comment by TheSisb** [comment, marked as answer]
Hi @zkuzmic,

Good catch, this isn't a desired behavior. We're open to PRs! We've been meaning to clean up this typing for a while and would greatly appreciate the assist!

**Comment by kate-porter** [comment]
Hey @zkuzmic! Thanks again for bringing this up. Just wanted to check in to see if you did want to open a PR to address this. If so, let us know if there's any support we can provide!

**Reply by zkuzmic** [reply]
I'd like to, just haven't gotten around to it yet, sorry 😅

**Reply by andioneto** [reply]
@zkuzmic All good! We just created a ticket on our Jira Board to address this. If you get to it sooner, let us know. We will follow up with you here when it is resolved. 🥇
